miniDAS/format.py

```diff
@@ -0,0 +1,161 @@
+from __future__ import annotations
+
+import logging
+from dataclasses import asdict, dataclass, field
+from pathlib import Path
+from typing import Any, Literal
+
+import h5py
+import numpy as np
+
+DataUnit = Literal["m/m", "cm/m", "mm/m"]
+ScaledUnit = Literal["ue/m"]
+_ALLOWED_DTYPES = (np.int32, np.float32)
+
+
+class OutOfBoundsError(BaseException):
+    ...
+
+
+@dataclass
+class Meta:
+    data_unit: DataUnit
+    start_time_ns: int
```

Comment:
`start_time_ns`, a Python `int` is like `long long int`, which equals to `int128` but there is some magic in CPython that manages int data type.